setup.py

```diff
@@ -341,11 +362,25 @@ def get_sunos_update():
 else:
     extensions = [ext]
 
+cmdclass = {}
+if py_limited_api:
+    class bdist_wheel_abi3(bdist_wheel):
+        def finalize_options(self):
+            bdist_wheel.finalize_options(self)
+            self.root_is_pure = False
+
+        def get_tag(self):
+            python, abi, plat = bdist_wheel.get_tag(self)
+            return python, "abi3", plat
+
+    cmdclass["bdist_wheel"] = bdist_wheel_abi3
```

Comment:
This looks like an hack. I guess this means bdist_wheel does not support Py_LIMITED_API natively? Should we keep an eye on this in the future? (e.g. the original class changes and this no longer works)

Reply:
Here's the example linked from the cibuildwheel docs:

Reply:
> I guess this means bdist_wheel does not support Py_LIMITED_API natively? Should we keep an eye on this in the future? (e.g. the original class changes and this no longer works)

`wheel` only has static support for Py_LIMITED_API through config file as far as I know.
It's probably worth keeping an eye on this in the future.

> which you probably want to do too

Well, I can't remember where I saw a recommendation against building cp3x abi3 with a more recent CPython than 3.x but I'm pretty sure I saw it and there was a rationale behind it. It might not be important in this case but there's probably no harm letting this as implemented now.

> Is manually setting root_is_pure necessary?

Indeed it's not.

Reply:
> Well, I can't remember where I saw a recommendation against building cp3x abi3 with a more recent CPython than 3.x but I'm pretty sure I saw it and there was a rationale behind it. It might not be important in this case but there's probably no harm letting this as implemented now.

I meant checking the implementation before setting the tag as `abi3` (i.e. `if python.startswith("cp"):`).
Now I see it's all wrapped with `if py_limited_api:` which is CPython-only, so this part of my comment is irrelevant. Whoops 😛

Indeed I think the official docs advise building cp3x ABI3 wheels using cp3x :)
Not sure what's the correct thing to do with the `bdist_wheel` class given the explicit setting of `Py_LIMITED_API = 0x03060000`, but I think either way should be fine here haha